utils/exceptions.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <QException>
+#include <QDebug>
+
+namespace jbrush
+{
+// Inherits from QException and allows the client to specify a unique string message that
+// is displayed in Qt's debugging console when raised
+class GeneralException : public QException
+{
+protected:
+    QString message;
+public:
+    GeneralException() { message = "Unidentified exception ocurred"; }
+    GeneralException(QString m) { message = m; }
+    void raise() const override
+    {
+        qDebug() << message;
+        throw *this;
+    }
+    QString errorMessage () const { return message; }
```

Comment:
This should just make an critical message box so that everyone can use it. OR even better, make two separate functions: one for ``qDebug()`` and one for ``QMessageBox::critical``.
```suggestion
    QString errorMessage () const { return message; }
    void errorWindow() const {QMessageBox::critical(nullptr, "Error", message);}
```